packages/core/src/dialects/db2/data-types.ts

```diff
@@ -116,16 +114,6 @@ export class DATE extends BaseTypes.DATE {
       this.options.precision = 6;
     }
   }
-
-  toSql() {
-    return `TIMESTAMP${this.options.precision != null ? `(${this.options.precision})` : ''}`;
-  }
-
-  toBindableValue(date: AcceptedDate) {
-    date = dayjs(date).utc(false);
-
-    return date.format('YYYY-MM-DD HH:mm:ss.SSS');
```

Comment:
this dialect does not support applyTimezone, so this should not inherit from the abstract version

Reply:
DB2 is not the only db that doesn't support applyTimezone.
Could we not just add a check to the applyTimezone function like the below:

```ts
  protected _applyTimezone(date: AcceptedDate) {
    const timezone = this._getDialect().sequelize.options.timezone;
    const supportsTimezone = this._getDialect().supports.globalTimeZoneConfig;

    if (timezone && supportsTimezone) {
      if (isValidTimeZone(timezone)) {
        return dayjs(date).tz(timezone);
      }

      return dayjs(date).utcOffset(timezone);
    }

    return dayjs(date);
  }
```

Reply:
After some testing, it looks like I do need to revert this.